src/main/core/manager.c

```diff
@@ -646,6 +653,55 @@ static void _manager_heartbeat(Manager* manager, SimulationTime simClockNow) {
     }
 }
 
+static void _manager_checkResourceUsage(Manager* manager) {
+    if (manager->checkFdUsage) {
+        unsigned long fd_count = 0;
+        DIR* dirp = opendir("/proc/self/fd");
+        if (dirp != NULL) {
+            struct dirent* entry = NULL;
+            while ((entry = readdir(dirp)) != NULL) {
+                fd_count++;
+            }
+        } else {
+            warning("Unable to open '/proc/self/fd': %s", g_strerror(errno));
+            manager->checkFdUsage = false;
+            return;
+        }
+
+        struct rlimit fd_lim;
+        if (getrlimit(RLIMIT_NOFILE, &fd_lim) != 0) {
+            warning("Unable to get fd limit: %s", g_strerror(errno));
+            manager->checkFdUsage = false;
+            return;
+        }
+
+        if (fd_count > fd_lim.rlim_cur * 90 / 100) {
+            warning("Using more than 90%% (%ld/%ld) of available file descriptors", fd_count,
+                    fd_lim.rlim_cur);
+            manager->checkFdUsage = false;
+        }
+    }
+
+    if (manager->checkMemUsage) {
+        long page_size = sysconf(_SC_PAGESIZE);
+        long avl_pages = sysconf(_SC_AVPHYS_PAGES);
+
+        if (page_size < 0 || avl_pages < 0) {
+            warning("Could not get memory usage information");
+            manager->checkMemUsage = false;
+            return;
+        }
+
+        long long avl_mem = page_size * avl_pages;
```

Comment:
I believe this will multiply two `long`s, truncating to a `long`, before assigning to the `long long`. I think what you mean is: `long long avl_mem = (long long)page_size * avl_pages;`

Also, probably clearer to use an explicit number of bits, such as int64_t.

Reply:
Good catch! And updated the type.